include/ByteBuffer.hpp

```diff
@@ -407,6 +407,14 @@ class ByteBuffer {
     int32_t capacity() {
         return m_capacity;
     }
+
+    bool operator!=(const ByteBuffer& other) {
```

Comment:
Implementing `operator!=` but not `operator==` is unsafe. You'll probably run into situations where `a == b` is true but `a != b` is also true. Instead, you can implement `operator==` and simply negate the result for `operator!=`.